whats4/src/components/FormMensagem.js

```diff
@@ -0,0 +1,181 @@
+import React from "react";
+import styled from "styled-components";
+
+const BoxPai = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+const BoxMensagem = styled.div`
+  display: flex;
+  flex-direction: column-reverse;
+  height: 100vh;
+  width: 45vw;
+  border: 1px solid black;
+  background-color: #e5ddd5;
+`;
+
+const DivMensagem = styled.div`
+  display: flex;
+  justify-content: space-evenly;
+  align-items: flex-end;
+  padding-bottom: 10px;
+`;
+
+const ListaMensagem = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  padding-left: 25px;
+  padding-bottom: 25px;
+`;
+
+const InputUsuario = styled.input`
+  height: 5vh;
+  width: 7vw;
+  border-radius: 5px;
+  border: none;
+  padding-left: 10px;
+  font-size: 16px;
+`;
+
+const InputMensagem = styled.input`
+  width: 28vw;
+  height: 5vh;
+  border-radius: 5px;
+  border: none;
+  padding-left: 10px;
+  font-size: 16px;
+`;
+
+const Botao = styled.button`
+  font-size: 16px;
+  font-weight: bold;
+  min-width: 6vw;
+  height: 5.2vh;
+  border-radius: 5px;
+  border: none;
+  background-color: white;
+`;
+
+const TextoBold = styled.span`
+  font-weight: bold;
+`;
+
+const MensagemRecebida = styled.div`
+  text-align: left;
+  background-color: white;
+  border-radius: 5px;
+  border: none;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  height: 7.5vh;
+  padding: 10px;
+  min-width: 7vw;
+  margin-bottom: 2vh;
+`;
+
+export class Input extends React.Component {
+  state = {
+    mensagens: [
+      {
+        usuario: "",
+        mensagem: "",
+      },
+    ],
+
+    valorInputUsuario: "",
+    valorInputMensagem: "",
+  };
+
+  adicionaMensagem = () => {
+    const novaMensagem = {
+      usuario: this.state.valorInputUsuario,
+      mensagem: this.state.valorInputMensagem,
+    };
+
+    const novasMensagens = [...this.state.mensagens, novaMensagem];
+
+    this.setState({
+      mensagens: novasMensagens,
+    });
+    this.setState({
+      valorInputMensagem: "",
+    });
```

Comment:
Aqui, vcs podem usar apenas um this.seState() para mudar todas as propriedades do estado de uma vez só. Otimiza o código e deixa ele mais limpo:

```
this.setState({ 
  mensagens: novasMensagens,
  valorInputMensagem: "",
})
```